### PR TITLE
PR for All Outputs: automating-tasks, checking-repository-health, set-up-continuous-integration, setting-up-linters, setting-up-your-repository, solving-the-problem, using-design-patterns, writing-tests, writing-html-js-css, delivering the product, reflecting on the learning-experience, setting up software development environment, getting peer review

### PR DESCRIPTION
**Pull request by johnkingzy**
#### What does this PR do?
- Fulfills the following outputs.
1. Writing-HTML-JS-CSS
2. Writing-tests
3. Design Patterns
4. Solving the Problem
4. Setting-up-your-repository
4. Setting-up-linters
6. Setting-up-continuous-integration
4. Checking-repository-health
5. Automating Tasks
6. Delivering your Product (Release Notes).
7. Reflecting on the learning-experience
8. Getting Peer reviews.
9. Setup software development environment.

#### Description of Task to be completed?
- Deliver a Folder containing the template for the proposed Front-end of your project.
- Deliver a Folder that Contains Tests Files and Dependencies
- Contain the Design Patterns that's been used in project.
- Contain the Class Implementation.
- Contain the link to my github repo for the proposed project
- Deliver two screenshots of ESLint and HoundCi
- Deliver a Link to a Successful Travis Build
- Deliver a Screenshot of my code coverage and GPA badges
- Deliver a screencast showing gulp watching and refreshing my browser
- Deliver a release notes for your final product in your repo's wiki.
- Deliver a Medium Post on my learning experience while producing these outputs.
- Deliver a screenshot of at least two peer code reviews
- Get at least two peer code reviews
- Delivers live URLs to a `review` app, your `staging` & `production` apps

#### How should this be manually tested?
- Run locally => Clone repository then install all dependencies using ``` npm install ```
- Pull request link includes the peer code review
- View the file `invertedIndex.js` to view class implementation.
- Visit the [app link](https://maximuf-index.herokuapp.com/) on Heroku

### CODE REVIEW

File: 06-building-a-program-start-to-finish/checking-repository-health/output/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: `Code Climate` and `Coveralls`
+- **Deliver**: a screenshot of your code coverage and GPA badges in your repo's README file
+- **With**: *independently*
+
+#### ScreenShot for my code coverage
+- This folder contains the screenshots for my code coverage and GPA badges
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/checking-repository-health/output/README.md:7]: Cool

File: 06-building-a-program-start-to-finish/automating-tasks/output/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#### Output
+- **Using**: Gulp
+- **Deliver**: a screencast showing gulp watching and refreshing your browser when changes are made to your HTML, CSS & JS files
+- **With**: *independently*
+
+#### Link to YOUTUBE VIDEO
+[Here is the Link to the Video](https://youtu.be/0O82rCCwu-4 "Automating Task using Gulp and Browser-Sync - Browser-reload")
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/automating-tasks/output/README.md:8]: Cool

File: 06-building-a-program-start-to-finish/delivering-your-product/output/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#### Output
+- **Using**: Google Doc
+- **Deliver**: `Release notes` for your final product in your repo's wiki
+- **With**: *independently*
+
+## Link to Release Note on Google Docs
+
+- [Preview on Google Doc](https://docs.google.com/document/d/1Qdm7NfE8XWAS1TVRkXJCbhDsDkmu7yNr9CUBXUxNMg4/edit)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/delivering-your-product/output/README.md:8]: Nice. You are however lacking a version number.

File: 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#### Output
+- **Using**: GitHub
+- **Deliver**: a Pull Request showing at least two Peer Code Reviews
+- **With**: *independently*
+
+#### Screenshots for the pull requests
+- [Peer Reviews]
+This folder contains the screenshots for the peer code review.
+
+[Here is the Link to the Pull Request](https://github.com/andela-ksolomon/invertedIndex/pull/6 "#10001 edit gulpfile for travis build ")
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md:10]: Hope you were able to implement the feedback you received.

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#### Output
+- **Using**: Medium
+- **Deliver**: a post reflecting on your learning experience while producing these outputs
+- **With**: *independently*
+
+#### Link to mu Medium Post 
+
+- [Medium Post](https://medium.com/@maximufkingzy/the-adventure-experience-so-far-andela-3c3f6ccbcfa1)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md:8]: Cool

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: TravisCI
+- **Deliver**: a link to a successful TravisCI build of your `development` branch (it should include test runs)
+- **With**: *independently*
+
+#### Link to a Successful Travis Build
+[Link to a Successful travis build](https://travis-ci.org/andela-ksolomon/invertedIndex)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md:7]: Cool

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+- **Deliver**: live URLs to a `review` app, your `staging` & `production` apps
+- **With**: *independently*
+
+#### Deliver live Urls
+##Review App:
+https://staging-maximuf-index-pr-6.herokuapp.com/
+##Staging App:
+https://staging-maximuf-index.herokuapp.com/
+##Production
+https://maximuf-index.herokuapp.com/
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md:12]: Application error on [Production](https://maximuf-index.herokuapp.com/) why?

File: 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: GitHub
+- **Deliver**: the repository for your proposed project and at least one `Pull Request`
+- **With**: *independently*
+
+#### Link To my Github repo
+[my github repo](https://github.com/andela-ksolomon/invertedIndex/tree/developing)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md:7]: Cool

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+   * @returns {Array} contains cleaned words
+   * cleans the keyword for search
+   */
+  static cleanValues(word) {
+    let value = word.replace(/[&\\#,+()$~%.'":*?<>{}]/g, '')
+      .replace(/[[|\]]/g, '')
+      .toLowerCase()
+      .trim()
+      .split(/\b\s+(?!$)/);
+    value = value.filter(word => word !== '');
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js]: 'word' is already declared in the upper scope. Please fix this.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+    expect(invalidKey).toThrowError(`${name} is not well formatted`);
+  });
+
+  it('Should return throw an error for Invalid file Extension', () => {
+    const validate = () => {
+      InvertedIndex.validateFile('result', 'sample.txt');
+    };
+    const error = 'sample.txt has an Invalid File extension, JSON only';
+    expect(validate).toThrowError(error);
+  });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: This test specification seems as though it should belong in your validateFile() Test Suite.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+    expect(checkEmptyBook).toThrowError();
+  });
+
+  it('Should throw an error for Invalid keys', () => {
+    const name = 'invalidBook.json';
+    const invalidKey = () => {
+      InvertedIndex.validateFile(invalidKeys, name);
+    };
+    expect(invalidKey).toThrowError(`${name} is not well formatted`);
+  });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: This test specification seems as though it should belong in your validateFile() Test Suite.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+      done();
+    });
+  });
+  it('Should throw an error for empty books', () => {
+    const fileName = 'emptybook.json';
+    const checkEmptyBook = () => {
+      InvertedIndex.validateFile(emptyBook, fileName);
+    };
+    expect(checkEmptyBook).toThrowError();
+  });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: This test specification seems as though it should belong in your validateFile() Test Suite.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+    const alltoken = Object.keys(getIndex);
+    expect(alltoken[0]).toBe('alice');
+  });
+});
+
+describe('InvertedIndex readFile', () => {
+  it('Should return true for well formatted File', () => {
+    const book = JSON.stringify(myBook);
+    expect(InvertedIndex.validateFile(book, 'books.json')).toBeTruthy();
+  });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: This test specification seems as though it should belong in your validateFile() Test Suite.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+    expect(numOfBooks.length).toEqual(6);
+  });
+});
+
+describe('InvertedIndex searchIndex', () => {
+  it('Should return `No file has been indexed yet', () => {
+    const search = () => {
+      invertedIndex.searchIndex('alice');
+    };
+    expect(search).toThrowError();
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: What error is thrown?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+  });
+
+  it('Should return `please enter a keyword to search.`', () => {
+    const search = () => {
+      invertedIndex.searchIndex();
+    };
+    expect(search).toThrowError('please enter a keyword to search.');
+  });
+
+  it('Should return', () => {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: should return what please?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+    expect(invertedIndex.getResult('alice', ['books.json']))
+      .toEqual({ alice: [0, 3] });
+  });
+
+  it('Should return', () => {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: should return what please?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+    const validInput = () => {
+      invertedIndex.getResult('[[[[[[[[[[[]]]]]]]]]', ['books.json']);
+    };
+    expect(validInput).toThrow('Search for Aplhanumeric values only');
+  });
+
+  it('Should return an array of clean values', () => {
+    expect(InvertedIndex.cleanValues('How, are, you doing today'))
+      .toEqual(['how', 'are', 'you', 'doing', 'today']);
+  });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Should this be part of your searchIndex() test suite?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 18]
```diff
@@ -0,0 +1,147 @@
...
+
+  it('Should return throw an error for Invalid file Extension', () => {
+    const validate = () => {
+      InvertedIndex.validateFile('result', 'sample.txt');
+    };
+    const error = 'sample.txt has an Invalid File extension, JSON only';
+    expect(validate).toThrowError(error);
+  });
+});
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: You seem to be lacking a test for your tokenize method. Please add it.

**Review by andela-moe — COMMENTED** [review]
Please address comment

File: 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md (hunk 4 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md:10]: Yes i did

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md (hunk 7 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md:12]: Yeah its has been fixed

File: 06-building-a-program-start-to-finish/delivering-your-product/output/README.md (hunk 3 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/delivering-your-product/output/README.md:8]: Thanks, it has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 14 above)
Comment (andela-moe) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Have you attended to this?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 19]
```diff
@@ -0,0 +1,159 @@
...
+  });
+
+  it('Should contain the tokenize method', () => {
+    expect(typeof InvertedIndex.tokenize).toBe('function');
+  });
+
+  it('Should contain the validateFile method', () => {
+    expect(typeof InvertedIndex.validateFile).toBe('function');
+  });
+});
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Please add a line space here for better readability.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 20]
```diff
@@ -0,0 +1,159 @@
...
+describe('InvertedIndex createIndex', () => {
+  it('Should return true for creating Index', () => {
+    expect(invertedIndex.createIndex('books.json', myBook)).toBeTruthy();
+  });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: So when the index is not created, what happens?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 21]
```diff
@@ -0,0 +1,159 @@
...
+
+  it('Should return the numbers of books in a file', () => {
+    const numOfBooks = invertedIndex.booksIndex('books.json');
+    expect(numOfBooks.length).toEqual(6);
+  });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: How is this a test specification for your createIndex test suite?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 22]
```diff
@@ -0,0 +1,159 @@
...
+});
+
+describe('InvertedIndex searchIndex', () => {
+  it('Should return `No file has been indexed yet', () => {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: When should this test specification pass?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 23]
```diff
@@ -0,0 +1,159 @@
...
+    const search = () => {
+      invertedIndex.searchIndex('alice');
+    };
+    expect(search).toThrowError('Error: No file has been indexed yet');
+  });
+
+  it('Should return an empty if not found', () => {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Your test specifications are not descriptive enough to communicate what the test is all about.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 24]
```diff
@@ -0,0 +1,159 @@
...
+    invertedIndex.searchIndex('alive', ['books.json']);
+    expect(invertedIndex.finalResult).toEqual(
+      { 'books.json': Object({ alive: [] }) }
+      );
+  });
+
+  it('Should return `please enter a keyword to search.`', () => {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: When?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 25]
```diff
@@ -0,0 +1,159 @@
...
+    };
+    expect(search).toThrowError('please enter a keyword to search.');
+  });
+
+  it('Should return an object for searchResult', () => {
+    expect(invertedIndex.getResult('alice', ['books.json']))
+      .toEqual({ alice: [0, 3] });
+  });
+
+  it('Should throw an error', () => {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Just like that? Why is this error being thrown?

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 26]
```diff
@@ -0,0 +1,224 @@
...
+      } else {
+        searchResult[word] = [];
+      }
+    });
+    return searchResult;
+  }
+
+  /**
+   * @cleanValues method
+   * @param {string} word - word to clean
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js]: Your JSDoc says **_word_** but you are passing in **_words_**. Please fix this

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js (hunk 9 above)
Comment (andela-moe) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js]: Your line spacings for this method are not proper. Please fix.

**Review by andela-moe — COMMENTED** [review]
Please address comments

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 18 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 10 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 11 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js (hunk 9 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js (hunk 26 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 25 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 24 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 23 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 22 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 21 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 20 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been created

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 19 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 27]
```diff
@@ -207,17 +212,18 @@ export default class InvertedIndex {
...
    */
-static cleanValues(words) {
+  static cleanValues(words) {
     let value = words.replace(/[&\\#,+()$~%.'":*?<>\-^!@{}]/g, '')
       .replace(/[[|\]]/g, '')
       .toLowerCase()
       .trim()
       .split(/\b\s+(?!$)/);
-    value = value.filter(word => word !== '');
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js]: Please check the trailing spaces and indentations in this your cleanValues() method.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 28]
```diff
@@ -151,9 +157,11 @@ describe('InvertedIndex validateFile', () => {
...
 
-describe('InvertedIndex cleanValues', () => {
-  it('Should return an array of clean values', () => {
+describe('InvertedIndex cleanValues Method', () => {
+
+  it('Should return an array of clean keywords', () => {
     expect(InvertedIndex.cleanValues('How, are, you doing today'))
       .toEqual(['how', 'are', 'you', 'doing', 'today']);
   });
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Please go through your tests again. You have eslint violation errors (Max-len, Trailing spaces, Unnecessary line breaks after in your `describe` function just before your `it` function) and also ensure that your tests are all right.

**Review by andela-moe — COMMENTED** [review]
Please review comments

**Comment by andela-moe** [comment]
Also ensure that this PR is properly formatted. See your peers for what is expected.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 29]
```diff
@@ -0,0 +1,229 @@
+/* global FileReader */
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js]: This file's name should be in PascalCase.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js [hunk 30]
```diff
@@ -0,0 +1,229 @@
...
+    const value = words.replace(/[&\\#,+()$~%.'":*?<>\-^!@{}]/g, '')
+      .replace(/[[|\]]/g, '')
+      .toLowerCase()
+      .trim()
+      .split(/\b\s+(?!$)/)
+      .filter(word => word !== '');
+    return value;
+  }
+}
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js]: Nice work

File: 06-building-a-program-start-to-finish/using-design-patterns/output/README.md [hunk 31]
```diff
@@ -0,0 +1,38 @@
...
+  }
+  /**
+   * @createIndex method
+   * @param {string} fileName
+   * @param {Object} fileContent
+   * @returns {boolean} true - index created successfully
+   * create index of the fileName
+   */```
+
+To use the methods of the object you first have to create an instance of the class by doing: myClass = new InvertedIndex() then myclass.createIndex();
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/using-design-patterns/output/README.md:38]: Cool

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 32]
```diff
@@ -0,0 +1,169 @@
...
+  it('Should return true for creating Index', () => {
+    const indexedBooks = invertedIndex.indexedBooks,
+      fileName = 'books.json';
+    invertedIndex.createIndex(fileName, myBook);
+    expect(indexedBooks.includes(fileName)).toBeTruthy();
+  });
+});
+
+describe('InvertedIndex searchIndex', () => {
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: This line space is not necessary. Please fix it.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 33]
```diff
@@ -0,0 +1,169 @@
...
+  it('Should throw an error if keyword contains special characters', () => {
+    const validInput = () => {
+      invertedIndex.getResult('[[[[[[[[[[[]]]]]]]]]', ['books.json']);
+    };
+    expect(validInput).toThrow('Search for Aplhanumeric values only');
+  });
+});
+
+describe('InvertedIndex getIndex Method', () => {
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: This line space is not necessary. Please fix it.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 34]
```diff
@@ -0,0 +1,169 @@
...
+    const readFile = InvertedIndex.readFile(jsonFile);
+    readFile.then((res) => {
+      expect(res[1][0].title).toBe('Alice in Wonderland');
+      done();
+    });
+  });
+});
+
+describe('InvertedIndex tokenize Method', () => {
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: This line space is not necessary. Please fix it.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 35]
```diff
@@ -0,0 +1,169 @@
...
+  it('Should return an array of tokenize words', () => {
+    expect(InvertedIndex.tokenize('alice%,monday,&tuesday#, ]]]]['))
+    .toEqual(['alice', 'monday', 'tuesday']);
+  });
+});
+
+describe('InvertedIndex validateFile Method', () => {
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: This line space is not necessary. Please fix it.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 36]
```diff
@@ -0,0 +1,169 @@
...
+  });
+
+  it('Should return `alive` as the first token created', () => {
+    const alltoken = Object.keys(getIndex);
+    expect(alltoken[0]).toBe('alice');
+  });
+});
+
+describe('InvertedIndex readFile Method', () => {
+  it('Should return `Alice in Wonderland` as first title of the book', (done) => {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: This line exceeds 80 characters. Please fix this.

**Review by andela-moe — COMMENTED** [review]
Please review comments

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 35 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 34 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 36 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 33 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 32 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: thanks this has been updated

File: 06-building-a-program-start-to-finish/using-design-patterns/output/README.md (hunk 31 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/using-design-patterns/output/README.md:38]: Thanks

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js (hunk 30 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js]: Thanks

File: 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js (hunk 29 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 37]
```diff
@@ -0,0 +1,166 @@
...
+    const search = () => {
+      invertedIndex.searchIndex();
+    };
+    expect(search).toThrowError('please enter a keyword to search.');
+  });
+
+  it('Should return an object for the searchResult', () => {
+    expect(invertedIndex.getResult('alice', ['books.json']))
+      .toEqual({ alice: [0, 3] });
+  });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Should this be part of your searchIndex method test?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 38]
```diff
@@ -0,0 +1,166 @@
...
+
+  it('Should throw an error if keyword contains special characters', () => {
+    const validInput = () => {
+      invertedIndex.getResult('[[[[[[[[[[[]]]]]]]]]', ['books.json']);
+    };
+    expect(validInput).toThrow('Search for Aplhanumeric values only');
+  });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Should this be part of your searchIndex method test?

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 39]
```diff
@@ -0,0 +1,166 @@
+/* My Test Setups */
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Please use the proper naming convention for this test file. Meet your peers to inquire on what is preferred.

**Review by andela-moe — COMMENTED** [review]
Please review comments.

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 39 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 37 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 38 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks this has been updated

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js [hunk 40]
```diff
@@ -1,4 +1,4 @@
-/* My Test Setups */
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Rename this file. Please see your peers for the preferred naming standards for test files.

**Review by andela-moe — COMMENTED** [review]
Please review comments

File: 06-building-a-program-start-to-finish/writing-tests/output/spec.js (hunk 40 above)
Comment (johnkingzy) [reply 06-building-a-program-start-to-finish/writing-tests/output/spec.js]: Thanks i have reflected on this

**Pull request merged**